misk/src/main/kotlin/misk/config/MiskConfig.kt

```diff
@@ -0,0 +1,76 @@
+package misk.config
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.fasterxml.jackson.module.kotlin.MissingKotlinParameterException
+import misk.environment.Environment
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.net.URL
+
+object MiskConfig {
+  fun <T : Config> load(
+      configClass: Class<out Config>,
+      appName: String,
+      environment: Environment
```

Comment:
Since we're now requiring that the environment be known before injection, we should move the lookup logic from ``EnvironmentModule.fromEnvironmentVariable`` to ``Environment.fromEnvironmentVariable`` so that you can get the ``Environment`` without creating an injector.

e.g.

````kotlin
enum class Environment {
   companion object {
       fun fromEnvironmentVariable() : Environment { /* existing lookup logic */   }
   }
}

class EnvironmentModule(val environment: Environment) : KAbstractModule() {
    override fun configure() {
      bind<Environment>().toInstance(environment)
    }
}
````